skore/src/skore/sklearn/cross_validation/cross_validation_reporter.py

```diff
@@ -0,0 +1,165 @@
+
+
+class CrossValidationReporter:
+    """Evaluate estimator by cross-validation and output UI-friendly object.
+
+    This class wraps scikit-learn's :func:`sklearn.model_selection.cross_validate`
+    function, to provide more context and facilitate the analysis.
+    As such, the arguments are the same as the
+    :func:`sklearn.model_selection.cross_validate` function.
+
+    For a user guide and in-depth example, see :ref:`example_cross_validate` and
+    :ref:`example_track_cv`.
+
+    More precisely, this class does the following:
+
+    *   Detect the ML task being performed, based on the estimator and data
+
+    *   Based on the ML task, add appropriate metrics to compute during
+        cross-validation
+
+    *   Perform the cross-validation itself
+
+    *   Save the result to ``project``, if available
+
+    *   Clean the cross-validation results so that the output of the function is as
+        close as possible to scikit-learn's
+
+    *   Return the clean cross-validation results.
+
+    Parameters
+    ----------
+    estimator : estimator object implementing ‘fit’
+        The object to use to fit the data.
+
+    X : {array-like, sparse matrix} of shape (n_samples, n_features)
+        The data to fit. Can be for example a list, or an array.
+
+    y : array-like of shape (n_samples,) or (n_samples, n_outputs), default=None
+        The target variable to try to predict in the case of supervised learning.
+
+    groups : array-like of shape (n_samples,), default=None
+        See :func:`sklearn.model_selection.cross_validate`.
+
+    scoring : str, callable, list, tuple, or dict, default=None
+        See :func:`sklearn.model_selection.cross_validate`.
+
+    cv : int, cross-validation generator or an iterable, default=None
+        See :func:`sklearn.model_selection.cross_validate`.
+
+    n_jobs : int, default=None
+        See :func:`sklearn.model_selection.cross_validate`.
+
+    verbose : int, default=0
+        See :func:`sklearn.model_selection.cross_validate`.
+
+    params : dict, default=None
+        See :func:`sklearn.model_selection.cross_validate`.
+
+    pre_dispatch : int or str, default=’2*n_jobs’
+        See :func:`sklearn.model_selection.cross_validate`.
+
+    return_train_score : bool, default=False
+        See :func:`sklearn.model_selection.cross_validate`.
+
+    return_estimator : bool, default=False
+        See :func:`sklearn.model_selection.cross_validate`.
+
+    return_indices : bool, default=False
+        See :func:`sklearn.model_selection.cross_validate`.
+
+    error_score : 'raise' or numeric, default=np.nan
+        See :func:`sklearn.model_selection.cross_validate`.
+
+    Attributes
+    ----------
+    cv_results : dict
+        A dict of the form returned by scikit-learn's
+        :func:`sklearn.model_selection.cross_validate` function.
+
+    X : array-like
+        The data that was fitted.
+
+    y : array-like or None
+        The target variable, or None if not provided.
+
```

Comment:
You've omitted the plot (property and attributes are the same imo), is it deliberate?

Reply:
well spotted, solved